From: Michael Chan <michael.chan@broadcom.com>
Date: Mon, 12 Oct 2020 05:10:48 -0400
Subject: bnxt_en: Set driver default message level.
Patch-mainline: v5.10-rc1
Git-commit: 8fb35cd302f74e63db4ce43a44e5e5fae44d80e3
References: jsc#SLE-16649

Currently, bp->msg_enable has default value of 0.  It is more useful
to have the commonly used NETIF_MSG_DRV and NETIF_MSG_HW enabled by
default.

v2: Change the fall back bnxt_reset_task() inside bnxt_rx_ring_reset()
to silent mode.  With older fw, we would take the fall back path and
it would be very noisy.

Reviewed-by: Edwin Peer <edwin.peer@broadcom.com>
Reviewed-by: Vasundhara Volam <vasundhara-v.volam@broadcom.com>
Signed-off-by: Michael Chan <michael.chan@broadcom.com>
Link: https://lore.kernel.org/r/1602493854-29283-4-git-send-email-michael.chan@broadcom.com
Signed-off-by: Jakub Kicinski <kuba@kernel.org>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/net/ethernet/broadcom/bnxt/bnxt.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/net/ethernet/broadcom/bnxt/bnxt.c
+++ b/drivers/net/ethernet/broadcom/bnxt/bnxt.c
@@ -69,6 +69,7 @@
 #include "bnxt_debugfs.h"
 
 #define BNXT_TX_TIMEOUT		(5 * HZ)
+#define BNXT_DEF_MSG_ENABLE	(NETIF_MSG_DRV | NETIF_MSG_HW)
 
 static const char version[] =
 	"Broadcom NetXtreme-C/E driver " DRV_MODULE_NAME " v" DRV_MODULE_VERSION "\n";
@@ -10757,7 +10758,7 @@ static void bnxt_rx_ring_reset(struct bn
 			else
 				netdev_warn(bp->dev, "RX ring reset failed, rc = %d, falling back to global reset\n",
 					    rc);
-			bnxt_reset_task(bp, false);
+			bnxt_reset_task(bp, true);
 			break;
 		}
 		bnxt_free_one_rx_ring_skbs(bp, i);
@@ -12518,6 +12519,7 @@ static int bnxt_init_one(struct pci_dev
 		return -ENOMEM;
 
 	bp = netdev_priv(dev);
+	bp->msg_enable = BNXT_DEF_MSG_ENABLE;
 	bnxt_set_max_func_irqs(bp, max_irqs);
 
 	if (bnxt_vf_pciid(ent->driver_data))
